web/src/Setting.js

```diff
@@ -126,8 +127,9 @@ export function getUserAvatar(username, isLarge = false) {
     return null;
   }
 
-  let gravatarStr = md5(username);
-  return `https://www.gravatar.com/avatar/${gravatarStr}?d=retro`;
+  let identicon = new Identicon(md5(username), 420);
+  return "data:image/png;base64," + identicon;
+  // return `https://www.gravatar.com/avatar/${gravatarStr}?d=retro`;
```

Comment:
remove all comments here

Reply:
I hava solved the three problems you said before,thank you for your advices.